sdk/textanalytics/ai-text-analytics/src/transforms.ts

```diff
@@ -0,0 +1,346 @@
+}
+
+/**
+ * @internal
+ */
+export function transformActionResult(
+  actionName: AnalyzeActionName,
+  input: TextDocumentInput[] | LanguageDetectionInput[],
+  response: AnalyzeResponse
+):
+  | EntityLinkingResult[]
+  | EntityRecognitionResult[]
+  | KeyPhraseExtractionResult[]
+  | SentimentAnalysisResult[]
+  | LanguageDetectionResult[] {
+  switch (response.kind) {
+    case "EntityLinkingResults": {
+      return toEntityLinkingResult(input, (response as EntityLinkingTaskResult).results);
+    }
+    case "EntityRecognitionResults": {
+      return toEntityRecognitionResult(input, (response as EntitiesTaskResult).results);
+    }
+    case "KeyPhraseExtractionResults": {
+      return toKeyPhraseExtractionResult(input, (response as KeyPhraseTaskResult).results);
+    }
+    case "PiiEntityRecognitionResults": {
+      return toPiiEntityRecognitionResult(input, (response as PiiTaskResult).results);
+    }
+    case "SentimentAnalysisResults": {
+      return toSentimentAnalysisResult(input, (response as SentimentTaskResult).results);
+    }
+    case "LanguageDetectionResults": {
+      return toLanguageDetectionResult(input, (response as LanguageDetectionTaskResult).results);
+    }
+    default: {
+      throw new Error(`Unsupported action name: ${actionName}`);
+    }
+  }
+}
+
+function appendReadableErrorMessage(currentMessage: string, innerMessage: string): string {
+  let message = currentMessage;
+  if (message.slice(-1) !== ".") {
+    message = message + ".";
+  }
+  return message + " " + innerMessage;
+}
+
+/**
+ * @internal
+ * parses incoming errors from the service/
+ * @param error - the incoming error
+ */
+export function transformError(errorResponse: unknown): any {
+  const castErrorResponse = errorResponse as {
+    response: {
+      parsedBody?: ErrorResponse;
+    };
+    statusCode: number;
+  };
+  if (!castErrorResponse.response) {
+    throw errorResponse;
+  }
+  const topLevelError = castErrorResponse.response.parsedBody?.error;
+  if (!topLevelError) return errorResponse;
+  let errorMessage = topLevelError.message;
+  let code = topLevelError.code;
+  function unwrap(error: ErrorModel | InnerErrorModel): ErrorModel {
+    const innerError = error.innererror;
+    if (innerError) {
+      if (innerError.message) {
+        errorMessage = appendReadableErrorMessage(errorMessage, innerError.message);
+      }
+      if (innerError.code) {
+        code = innerError.code;
+      }
+      return unwrap(innerError);
+    }
+    return error as ErrorModel;
+  }
+  unwrap(topLevelError);
+  return new RestError(errorMessage, {
+    code,
+    statusCode: castErrorResponse.statusCode,
+  });
+}
+
+/**
+ * @internal
+ */
+export function setDefaultActionParameters<ActionName extends AnalyzeActionName>(
+  actionName: AnalyzeActionName,
+  action: AnalyzeActionParameters<ActionName>
+): AnalyzeActionParameters<ActionName> {
+  switch (actionName) {
+    case "PiiEntityRecognition": {
```

Comment:
Only PII has default action parameters? Does this override service-level defaults?

Reply:
Any action that works on sensitive data should have service-side logging off by default. Currently, only PII does this but in the next method I'll add, healthcare will also need the same treatment. The old swagger modeled this correctly by having unique definition of the parameter per action. On the other hand, the current swagger uses only one parameter definition for all the actions, so I opted to fix this behavior in the client instead.